### ISSUE TITLE
Resource view: Better way of displaying what is currently pinned 

### ISSUE THREAD

**Issue by Lindsayauchin** [concourse/concourse#4765]
![Screen Shot 2019-11-14 at 5 34 59 PM](https://user-images.githubusercontent.com/27746614/68901993-1febcf00-0705-11ea-9988-0f29b82a2322.png)

![Screen Shot 2019-11-14 at 5 35 40 PM](https://user-images.githubusercontent.com/27746614/68902019-32fe9f00-0705-11ea-8576-6af48c206796.png)

Related:
#3961 
#3169

**Comment by pivotal-jamie-klassen** [comment]
* don't show check bar when resource is pinned

**Comment by Lindsayauchin** [comment]
- expand version status bar to accommodate version length. 
- add expanded version of comment height
- width - truncated long version ref with ellipse at 50%
- expand row to show the whole comment

**Comment by Lindsayauchin** [comment]
## Updated Sketch:

![Screen Shot 2019-11-20 at 4 00 13 PM](https://user-images.githubusercontent.com/27746614/69277813-f15f6f80-0bae-11ea-8d8a-4bf83b2bf470.png)

**Comment by vito** [comment]
Splitting out the editable pin comment to https://github.com/concourse/concourse/issues/4781

**Comment by zoetian** [comment]
I'm still working on this right now:

It took longer than I think, since I'm carefully moving all the tests for the old "pin-bar" (which is the check bar we mentioned before) towards the new "pin-tools". There are a lot difference btw these two tests structures we have right now, so I'm still organizing this section.

Other than that, I also made some changes to the "last checked" sections but I don't want to mess up with the commit order, so it's currently stashed in my local. I'm going to push all my changes tonight (prob. the morning in EST time).  Sorry it took so long...

**Comment by zoetian** [comment]
Note: I noticed that there weren't any obvious indications when pin bar fails to pin/ unpin certain version via `pin-icon`. After some discussion with Lindsay, we decided to add an error state for pin bar/tools.
Other than that, the current code and tests are all passed.

I just still need to 
- add the error state for `when pinning fails` (add another model state??) AND
- display the red border pin-tools with crossed error icon AND
- proper error messages

**Comment by zoetian** [comment]
So as I'm adding new error state for both pinning & unpinning failures, I noticed that the parent component `pinTools`, which wraps both `pinBar` and `pinCommentBar` will eventually have multiple input from both child components. This will make the code really messy (i.e. when we're rendering the `pinTools` in body, it needs 2 `model` since `pinBar`'s signature requires it as a "extensive record". 
After reading this [article] (http://folkertdev.nl/blog/elm-child-parent-communication/) I decided to try  it out by adding a Msg onto the `pinTools` and return tuples from both `pinBar` and `pinCommentBar`. Maybe it will work in a more decent way? 🤔🤔🤔

**Comment by pivotal-jamie-klassen** [comment]
After spending the morning investigating this epic and the current state of the `issue/4765` branch, we have concluded the following:
* when the resource is pinned, the check status bar is replaced with a new "pin tools" widget
* pin tools correctly displays the pinned version and has a working unpin button
* the pin bar has already been removed, which feels reasonable (#4780)
* the comment overlay has already been removed (#4787), but this seems premature as the inline comment editing feature is currently broken:
<img width="1435" alt="Screen Shot 2020-03-16 at 11 54 52 AM" src="https://user-images.githubusercontent.com/22056320/76776332-fdce9000-677c-11ea-984c-f709d8d239a5.png">

Questions at this point: what is the "error state" feature? How does it work?  Is it complete?

The plan for now is to try to narrow the breadth of changes in the `issue/4765` branch, ideally issuing a PR by the end of today. In this PR, we expect:
* no inline comment functionality - the pin tools will only consist of the unpin button and pinned version display
* the comment overlay will remain in place for the time being
* we will assess the completeness of the error state functionality and defer deciding whether to include it in today's PR.

We intend to make this changes on the `issue/4765` branch, but we don't want the existing work on this branch to go to waste. Accordingly, we will create a duplicate of the current state of work on a new branch called `resource-ui-polish` - this should allow @zoetian to pick up exactly where she left off.
